run/types/DeviceWrapper.ts

```diff
@@ -1551,60 +1574,74 @@ export class DeviceWrapper implements SharedDeviceInterface {
   /* ======= Settings functions =========*/
 
   public async turnOnReadReceipts(platform: SupportedPlatformsType) {
-    if (platform === "android") {
-      await this.navigateBack(platform);
-      await sleepFor(100);
-      await this.clickOnByAccessibilityID("User settings");
-      await sleepFor(500);
-      await this.clickOnElementById(`network.loki.messenger:id/privacyButton`);
-      await sleepFor(2000);
-      await this.clickOnElementAll({
-        strategy: "id",
-        selector: "android:id/summary",
-        text: "Send read receipts in one-to-one chats.",
-      });
-      await this.navigateBack(platform);
-      await sleepFor(100);
-      await this.navigateBack(platform);
-    } else {
-      await this.navigateBack(platform);
-      await sleepFor(100);
-      await this.clickOnByAccessibilityID("User settings");
-      await this.clickOnElementAll({ strategy: "id", selector: "Privacy" });
-      await this.clickOnElementAll({
-        strategy: "xpath",
-        selector: `//XCUIElementTypeSwitch[@name="Read Receipts, Send read receipts in one-to-one chats."]`,
-      });
-      await this.navigateBack(platform);
-      await sleepFor(100);
-      await this.clickOnByAccessibilityID("Close button");
-    }
+    await this.navigateBack(platform);
+    await sleepFor(100);
+    await this.clickOnByAccessibilityID("User settings");
+    await sleepFor(500);
+    await this.clickOnElementAll(new PrivacyButton(this));
+    await sleepFor(2000);
+    await this.clickOnElementAll(new ReadReceiptsButton(this));
+    await this.navigateBack(platform);
+    await sleepFor(100);
+    await this.clickOnElementAll(new ExitUserProfile(this));
   }
 
   public async checkPermissions(platform: SupportedPlatformsType) {
-    let permissions;
-    if (platform === "ios") {
-      permissions = await this.doesElementExist({
-        strategy: "accessibility id",
-        selector: "Don’t Allow",
-        maxWait: 1000,
-      });
-    } else {
-      permissions = await this.doesElementExist({
+    if (platform === "android") {
```

Comment:
we are in the DeviceWrapper class, so we already have the platform from 
`this.isIOS()` & `this.isAndroid()`
you should be able to remove the `platform: SupportedPlatformsType` altogether

Reply:
Fixed